src/core_modules/capture-core/components/Pages/Search/SearchResults/SearchResults.component.js

```diff
@@ -48,13 +45,28 @@ const CardListButtons = withStyles(buttonStyles)(
         currentSearchScopeType,
         id,
         orgUnitId,
-        navigationButtonsState,
+        enrollmentType,
         programName,
         classes,
     }) => {
+        const deriveNavigationButtonState =
+          (type): $Keys<typeof availableCardListButtonState> => {
+              switch (type) {
+              case enrollmentTypes.ACTIVE:
+                  return availableCardListButtonState.SHOW_VIEW_ACTIVE_ENROLLMENT_BUTTON;
+              case enrollmentTypes.CANCELLED:
+              case enrollmentTypes.COMPLETED:
+                  return availableCardListButtonState.SHOW_RE_ENROLLMENT_BUTTON;
+              default:
+                  return availableCardListButtonState.DONT_SHOW_BUTTON;
+              }
+          };
+
+        const navigationButtonsState = deriveNavigationButtonState(enrollmentType);
+
```

Comment:
As I was explaining before now the buttons component are responsible of what buttons is to be shown